tests/common/fixtures/ptfhost_utils.py

```diff
@@ -177,7 +177,11 @@ def run_icmp_responder(duthost, ptfhost, tbinfo):
     templ = Template(open(os.path.join(TEMPLATES_DIR, ICMP_RESPONDER_CONF_TEMPL)).read())
     ptf_indices = duthost.get_extended_minigraph_facts(tbinfo)["minigraph_ptf_indices"]
     vlan_intfs = duthost.get_vlan_intfs()
+    vlan_table = duthost.get_running_config_facts()['VLAN']
+    vlan_name = list(vlan_table.keys())[0]
```

Comment:
There could be multiple vlans. In this case, we are going to have multiple MAC (one per vlan)

Reply:
Is there a way to differentiate between VLANs when receiving heartbeats on the PTF? Otherwise ICMP responder will not know which MAC to use.

Reply:
@tahmed-dev , my understanding is its the same mac for all Vlans for dual-tor case. Are you referring in general?